### PR TITLE
EDD-10: Send user to EDL to accept a EULA when necessary

### PR DESCRIPTION
**Pull request by macrouch**
# Overview

### What is the feature?

When EDD detects a EULA needs to be accepted, it will open the user's browser to EDL with a redirect_uri provided by the initiating application (EDSC). After that application opens EDD again, the download will try again.

This PR also adds some more useful information as errors when a file fails to download.

### What areas of the application does this impact?

Downloading files behind a EULA

# Testing

### Reproduction steps

Initiate a download from EDSC for files behind a EULA, with a user that has not accepted the EULA. After accepting the EULA EDL will redirect to EDSC, which will open EDD and continue downloading files

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have bumped the `version` field in package.json and ran `npm install`

### CODE REVIEW

**Comment by macrouch** [comment]
Add documentation for eulaCallback deep link

File: src/main/eventHandlers/cancelDownloadItem.ts [hunk 1]
```diff
@@ -28,6 +26,12 @@ const cancelDownloadItem = async ({
   }
 
   if (downloadId && !filename) {
+    // await database.updateFilesWhere({
```
Comment (eudoroolivares2016) [inline src/main/eventHandlers/cancelDownloadItem.ts:29]: remove?

File: src/main/eventHandlers/sendToEula.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    downloadId,
+    fileId: providedFileId,
+    forceLogin = false
+  } = info
+
+  let fileId = providedFileId
+
+  // If no fileId was provided, find the file that is waitingForEula for the downloadId
+  if (!providedFileId) {
+    ({ id: fileId } = await database.getFileWhere({
```
Comment (eudoroolivares2016) [inline src/main/eventHandlers/sendToEula.ts:31]: Can there be multiple files in the `waitingForEula` state here?

Comment (macrouch) [reply src/main/eventHandlers/sendToEula.ts:31]: There shouldn't be, but if there is `getFileWhere` only returns the first result

**Review by eudoroolivares2016 — APPROVED** [review]

**Review by dpesall — APPROVED** [review]

**Review by trevorlang — APPROVED** [review]

**Pull request merged**